source/Ship.cpp

```diff
@@ -687,6 +716,22 @@ void Ship::Save(DataWriter &out) const
 			out.Write("destination system", targetSystem->Name());
 		if(isParked)
 			out.Write("parked");
+		if(!cargo.IsEmpty())
+		{
+			out.Write("cargo hold");
+			out.BeginChild();
+			{
+				for(const auto &it : cargo.MissionCargo())
+					out.Write("mission", it.first->Name(), "cargo", it.second);
```

Comment:
The player can have multiple missions or jobs active that have the same name, so this is not a unique-enough identifier. (This is why I suggested a UUID be used in the linked bug report.)

Reply:
I'm working on this problem next.

Reply:
Done.

Random::UUID() makes UUIDs. When a mission is instantiated, it gets a UUID. There's code in place to give UUIDs to missions from save files that don't yet have UUIDs (backward compatibility to old saves). The cargo holds use UUID to track mission cargo and mission passengers.